.github/workflows/build-icu.yml

```diff
@@ -59,14 +59,194 @@ jobs:
+            -Dexec.args="-d target/cldr-prod/common/"
+          (cd target/cldr-prod/ && zip -r ../cldr-prod.zip *)
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       - name: Upload cldr-prod
         uses: actions/upload-artifact@v2
         with:
           name: cldr-prod
-          path: target/cldr-prod
+          path: target/cldr-prod.zip
+  genicudata:
+    name: Generate ICU Data
+    runs-on: ubuntu-latest
+    needs: proddata
+    steps:
+      - name: Download CLDR Production Artifact
+        id: downloadprod
+        uses: actions/download-artifact@v2
+        with:
+          name: cldr-prod
+          path: .
+      - name: Download CLDR tools
+        id: downloadtools
+        uses: actions/download-artifact@v2
+        with:
+          name: cldr-code
+          path: cldr-code
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Setup Maven Settings
+        uses: s4u/maven-settings-action@v2.4.1
+        with:
+          servers: |
+            [{
+                "id": "github",
+                "username": "${{ github.actor }}",
+                "password": "${{ github.token }}"
+            },{
+                "id": "githubicu",
+                "username": "${{ github.actor }}",
+                "password": "${{ github.token }}"
+            },{
+                "id": "githubcldr",
+                "username": "${{ github.actor }}",
+                "password": "${{ github.token }}"
+            }]
+      - name: Create empty CLDR dir, unpack cldr-prod
+        run: mkdir -p cldr cldr-prod && cd cldr-prod && unzip ../cldr-prod.zip
+      # Don't actually need CLDR
+      # - name: Clone CLDR
+      #   uses: actions/checkout@v2
+      #   with:
+      #     ref: ${{ github.event.inputs.cldr-ref }}
+      #     lfs: false
+      #     repository: ${{ github.event.inputs.cldr-repo }}
+      #     path: cldr
+      - name: Clone ICU
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ github.event.inputs.icu-repo }}
+          ref: ${{ github.event.inputs.icu-ref }}
+          lfs: false # not needed here
+          path: icu
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('icu/tools/**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: Install CLDR tools
+        # TODO: Should not have to do this, see ICU-21425
+        # Do this after restoring cache.
+        run: >
+          mvn -B install:install-file   -Dproject.parent.relativePath=""   -DgroupId=org.unicode.cldr
+          -DartifactId=cldr-api   -Dversion=0.1-SNAPSHOT   -Dpackaging=jar   -DgeneratePom=true
+          -Dfile=cldr-code/cldr-code.jar
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Generate ICU Data
+        run: >
+          env
```

Comment:
In the [cldr-icu-readme.txt](https://github.com/unicode-org/icu/blob/main/icu4c/source/data/cldr-icu-readme.txt#L125) file I noticed that it sets other environment vars, like `ANT_OPTS="-Xmx4096m"` and `ICU4C_DIR`/`ICU4J_ROOT`.

I guess they might not actually be needed?

(FWIW, I think the `ICU4C_DIR` will be auto-set [here](https://github.com/unicode-org/icu/blob/7851f708a3dd43fc1e0e9dd784e7dfe9a70915b2/icu4c/source/data/build.xml#L25), but I'm not sure about `ICU4J_ROOT`)

Reply:
Didn't seem to be needed.

Reply:
`ICU4J_ROOT` is for the icu4j data install step (C Makefile) and is set on line 230, but is not needed here on line 147 - this is cldr-to-icu (David's tool)

Reply:
Ah, sorry for missing that. Thanks!